includes/class-sensei-teacher.php

```diff
@@ -811,6 +814,33 @@ public function filter_queries( $query ) {
 		}
 	}
 
+	/**
+	 * Filter the learners query to show
+	 * only users that belong to the current logged teacher.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @param string $sql The learners SQL query.
+	 *
+	 * @return string
+	 */
+	public function filter_learners_query( string $sql ): string {
+		if ( ! $this->is_admin_teacher() ) {
+			return $sql;
+		}
+
+		$teacher_learner_ids = $this->get_learner_ids_for_courses_with_edit_permission();
+		if ( ! $teacher_learner_ids ) {
+			$teacher_learner_ids = [ 0 ]; // Show no learners.
+		}
+
+		return str_replace(
+			'WHERE 1=1',
+			'WHERE 1=1 AND u.ID IN (' . implode( ',', array_map( 'absint', $teacher_learner_ids ) ) . ')',
```

Comment:
Thanks for pointing this out. I agree, this could be an issue in some cases. The default query size (max_allowed_packet) on newer MySQL versions is 16 MB, but on older ones, it is only 1 MB. I will check if this can be done with a join.

Reply:
Done! Could you please take another look? 🙏 

I've also added HPPS support. 😄 cc @merkushin